runtime-modules/content-directory/src/lib.rs

```diff
@@ -1964,6 +1954,50 @@ impl<T: Trait> Module<T> {
         Ok(())
     }
 
+    /// Ensure provided indices of `existing_properties`  are valid indices of `Class` properties
+    pub fn ensure_schema_properties_are_valid_indices(
+        existing_properties: &BTreeSet<PropertyId>,
+        class_properties: &[Property<T>],
+    ) -> dispatch::Result {
+        let has_unknown_properties = existing_properties
+            .iter()
+            .any(|&prop_id| prop_id >= class_properties.len() as PropertyId);
+        ensure!(
+            !has_unknown_properties,
+            ERROR_CLASS_SCHEMA_REFERS_UNKNOWN_PROP_INDEX
+        );
+        Ok(())
+    }
+
+    // Create new `Schema` with existing and new properies provided
+    pub fn create_class_schema(
+        existing_properties: BTreeSet<PropertyId>,
+        class_properties: &[Property<T>],
+        new_properties: &[Property<T>],
+    ) -> Schema {
+        // Create new Schema with existing properies provided
+        let mut schema = Schema::new(existing_properties);
+
+        // Add new property ids to `Schema`
+        new_properties.iter().enumerate().for_each(|(i, _)| {
+            let prop_id = (class_properties.len() + i) as PropertyId;
+
+            schema.get_properties_mut().insert(prop_id);
+        });
+        schema
+    }
+
+    /// Update existing `Class` properties with new ones provided, return updated ones
+    pub fn update_class_properties(
```

Comment:
No need to have a mutable input parameter, even if its mutating a copy, do this as a pure function.

Reply:
Did you mean, i should move ownership to the local mutable variable here?

Reply:
Done https://github.com/iorveth/substrate-runtime-joystream/commit/88eb6bebabad8c23d4c16127b9bf4c3abbda52a2